cedar-policy-validator/src/schema_file_format.rs

```diff
@@ -92,19 +133,37 @@ impl SchemaFragment {
 #[cfg_attr(feature = "wasm", tsify(into_wasm_abi, from_wasm_abi))]
 pub struct NamespaceDefinition {
     #[serde(default)]
-    #[serde(with = "::serde_with::rust::maps_duplicate_key_is_error")]
+    #[serde(deserialize_with = "deserialize_common_types")]
     #[serde(rename = "commonTypes")]
-    pub common_types: HashMap<SmolStr, SchemaType>,
+    pub common_types: HashMap<Id, SchemaType>,
     #[serde(rename = "entityTypes")]
-    #[serde(with = "::serde_with::rust::maps_duplicate_key_is_error")]
-    pub entity_types: HashMap<SmolStr, EntityType>,
+    #[serde(deserialize_with = "deserialize_entity_types")]
+    pub entity_types: HashMap<Id, EntityType>,
```

Comment:
same here, are we still detecting duplicate keys and returning an error?

Reply:
Please see the comment above.